include/vcpkg/base/system.proxy.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <optional>
+#include <string>
+
+/*
+ * This is a helper class. It reads IE Proxy settings.
+ * For coherence of vcpkg design "using HTTP(S)_PROXY not WPAD", it is a user-friendly design to manually
+ * read the IE Proxy settings on Windows, and set the environment variables HTTP(S)_PROXY to the same
+ * when starting an external program (like CMake, which file(DOWNLOAD) only accept proxy settings
+ * by HTTP(S)_PROXY variables).
+ */
+
+namespace vcpkg::System
+{
+    bool get_windows_ie_proxy_enabled();

```

Comment:
I don't think we should have this API; callers should call get_windows_ie_proxy_server and check that it's disengaged instead.

Reply:
Agree.